src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImageFunctionalTest.groovy

```diff
@@ -54,12 +62,130 @@ class DockerSaveImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
+
+        where:
+            image << [IMAGE_3_3, IMAGE_3_4]
+    }
+
+    def "saves all Docker images of specified base"() {
+        buildFile << pullImageTask('pullImage', IMAGE_3_4)
+        buildFile << pullImageTask('pullImage2', IMAGE_3_3)
+        buildFile << pullImageTask('pullImage3', IMAGE_3_2)
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn tasks.withType(DockerPullImage)
+                image = "${IMAGE_BASE}"
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            build('saveImage')
+
+        then:
+            def repoTags = getRepoTags(IMAGE_FILE)
+            repoTags.containsAll(IMAGE_3_2, IMAGE_3_3, IMAGE_3_4)
+            repoTags.every { it.startsWith("${IMAGE_BASE}:") }
+    }
+
+    def "only saves specific Docker images"() {
+        buildFile << pullImageTask('pullImage', IMAGE_3_4)
+        buildFile << pullImageTask('pullImage2', IMAGE_3_3)
+        buildFile << pullImageTask('pullImage3', IMAGE_3_2)
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn tasks.withType(DockerPullImage)
+                images.addAll("${IMAGE_3_3}", "${IMAGE_3_4}")
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            build('saveImage')
+
+        then:
+            getRepoTags(IMAGE_FILE) == [IMAGE_3_3, IMAGE_3_4]
+    }
+
+    def "saves specific Docker images from deprecated and current property"() {
+        buildFile << pullImageTask('pullImage', IMAGE_3_4)
+        buildFile << pullImageTask('pullImage2', IMAGE_3_3)
+        buildFile << pullImageTask('pullImage3', IMAGE_3_2)
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn tasks.withType(DockerPullImage)
+                image = "${IMAGE_3_3}"
+                images.add("${IMAGE_3_4}")
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            build('saveImage')
+
+        then:
+            getRepoTags(IMAGE_FILE) == [IMAGE_3_3, IMAGE_3_4]
+    }
+
+    def "skips execution if images is set to null"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task saveImage(type: DockerSaveImage) {
+                images.set(null as Set)
+                destFile = file("${IMAGE_FILE}")
+            }
+        """
+
+        when:
+            def result = build('saveImage')
+
+        then:
+            result.task(":saveImage").outcome == SKIPPED
+            !new File(projectDir, IMAGE_FILE).exists()
+    }
+
+    def getRepoTags(imageFile) {
```

Comment:
Would it have been easier/less brittle to use Docker Java here to inspect the repo tags?

Reply:
Can Docker Java look inside the exported image file?
If it only can ask the Docker daemon what I suspect, then it wouldn't help.
The question is not whether the tags are present in the repo, the check is whether the correct tags are exported to the image file.
Thus the content of the image file needs to be checked.

Reply:
Good point. I think Docker Java can't do this. Thanks for clarifying.